contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -51,32 +52,40 @@ const (
 	Deleted  = FlagType("deleted")
 )
 
+var defaultRoles = []string{"*"}
+
 // A struct that describes a pod task.
 type T struct {
 	ID          string
 	Pod         api.Pod
-	Spec        Spec
+	Spec        *Spec
```

Comment:
About your Clone comment: we should add a comment: "Spec is read-only everywhere, with the exception of the pod scheduler"